generate-stackbrew-library.sh

```diff
@@ -1,51 +1,101 @@
 #!/bin/bash
-set -e
+#
+# Based on https://github.com/docker-library/elasticsearch/blob/master/generate-stackbrew-library.sh
+set -eu
 
-declare -A aliases
-aliases=(
-        [6.6]='6 latest'
-        [5.5]='5'
+declare -A aliases=(
+    [6.6]='6 latest'
+    [5.5]='5'
 )
 
+self="$(basename "$BASH_SOURCE")"
 cd "$(dirname "$(readlink -f "$BASH_SOURCE")")"
+extglob_old=$(shopt -p extglob||true)
+shopt -s extglob
+
+versions=( +([0-9])\.+([0-9])/ )
+eval "$extglob_old"
 
-versions=( */ )
 versions=( "${versions[@]%/}" )
-url='git://github.com/docker-solr/docker-solr'
 
-echo '# maintainer: Martijn Koster <mak-github@greenhills.co.uk> (@makuk66)'
-echo '# maintainer: Shalin Mangar <shalin@apache.org> (@shalinmangar)'
+# sort version numbers with highest first
+IFS=$'\n'; versions=( $(echo "${versions[*]}" | sort -rV) ); unset IFS
+
+# get the most recent commit which modified any of "$@"
+fileCommit() {
+	git log -1 --format='format:%H' HEAD -- "$@"
+}
+
+# get the most recent commit which modified "$1/Dockerfile" or any file COPY'd from "$1/Dockerfile"
+dirCommit() {
+	local dir="$1"; shift
+	(
+		cd "$dir"
+		fileCommit \
+			Dockerfile \
+			$(git show HEAD:./Dockerfile | awk '
+				toupper($1) == "COPY" {
+					for (i = 2; i < NF; i++) {
+						print $i
+					}
+				}
+			')
+	)
+}
+
+cat <<-EOH
+# this file is generated via https://github.com/docker-library/solr/blob/$(fileCommit "$self")/$self
+
+Maintainers: Martijn Koster <mak-github@greenhills.co.uk> (@makuk66),
+             Shalin Mangar <shalin@apache.org> (@shalinmangar)
+GitRepo: https://github.com/docker-solr/docker-solr.git
+EOH
+
+# prints "$2$1$3$1...$N"
+join() {
+	local sep="$1"; shift
+	local out; printf -v out "${sep//%/%%}%s" "$@"
+	echo "${out#$sep}"
+}
 
 for version in "${versions[@]}"; do
-        if [ ! -f "$version/Dockerfile" ]; then
-          continue
-        fi
+	commit="$(dirCommit "$version")"
+
+	fullVersion="$(git show "$commit":"$version/Dockerfile" | awk '$1 == "ENV" && $2 == "ELASTICSEARCH_VERSION" { gsub(/~/, "-", $3); print $3; exit }')"
+
+	rcVersion="${version%-rc}"
 
-	commit="$(git log -1 --format='format:%H' -- "$version")"
-	fullVersion="$(grep -m1 'ENV SOLR_VERSION' "$version/Dockerfile" | cut -d' ' -f3)"
-	
 	versionAliases=()
-	while [ "$fullVersion" != "$version" -a "${fullVersion%[.-]*}" != "$fullVersion" ]; do
+	while [ "$fullVersion" != "$rcVersion" -a "${fullVersion%[.-]*}" != "$fullVersion" ]; do
 		versionAliases+=( $fullVersion )
 		fullVersion="${fullVersion%[.-]*}"
 	done
-	versionAliases+=( $version ${aliases[$version]} )
-	
+	versionAliases+=(
+		$rcVersion
+		${aliases[$version]:-}
+	)
+
 	echo
-	for va in "${versionAliases[@]}"; do
-		echo "$va: ${url}@${commit} $version"
-	done
+	cat <<-EOE
+		Tags: $(join ', ' "${versionAliases[@]}")
+		Architectures: amd64, arm32v5, arm32v7, arm64v8, i386, ppc64le, s390x
```

Comment:
Just a note that you'll want to make sure to update this if the `Architectures` of the `openjdk:8-jre` image ever change, or if the `FROM` of `solr` ever changes. :+1:

(We usually get a little crazy and scrape that directly via the `bashbrew` tool as in https://github.com/docker-library/httpd/blob/d8a2ea0c867407d2ecd335ee7836b31019cd484d/generate-stackbrew-library.sh#L36-L50, but that's definitely not necessary.  Just making sure you're aware that you'll need to manually ensure this list is correct. :+1:)

Reply:
Oh that's neat. I'm very much not interested in manual verification and having to remember things. I'll update to that. That means the script has a bashbrew dependency now, but that's not a big deal.
Running the new script I see that in fact the list I had was incorrect; in particular the alpine images are only for amd64. But I see slim versions, so I guess I can do another variant.